dev/tasks/homebrew-formulae/github.macos.yml

```diff
@@ -28,6 +28,8 @@ jobs:
 
       {{ macros.configure_homebrew_arrow(formula)|indent }}
       - name: Test formula
+        env:
+          HOMEBREW_NO_INSTALL_FROM_API: "1"
```

Comment:
Is this related? Can you add a comment explaining why this is required/desired?

Reply:
I should have extracted this as a separated task.
It seems that homebrew-cpp task is broken. We assume that it uses our configurations (`*.rb`) in https://github.com/apache/arrow/tree/main/dev/tasks/homebrew-formulae but they aren't used. I think that it's related to a Homebrew 4.0.0 change:

* https://brew.sh/2023/02/16/homebrew-4.0.0/
* https://docs.brew.sh/FAQ#can-i-edit-formulae-myself

I thought that this change fixes this but it seems that this change doesn't work...

I'll open an issue for it and revert this change from this PR.

Reply:
#40813